### ISSUE TITLE
SLDSTree: button needs discernible name 

### ISSUE THREAD

**Issue by jcjuett** [salesforce/design-system-react#1914]
Bug at this link: http://design-system-react-components.herokuapp.com/?selectedKind=SLDSTree&selectedStory=Base&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

Reproduce error: `<div class="slds-tree__item slds-is-selected"><button class="slds-button slds-button_icon slds-m-right_small slds-is-disabled" disabled="" tabindex="-1" type="button"><svg aria-hidden="true" class="slds-button__icon slds-button__icon_small"><use xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"></use></svg></button><span class="slds-size_1-of-1"><span class="slds-tree__item-label slds-truncate" title="Grains">Grains</span></span></div>`

Success criteria: add `><span class="slds-assistive-text">Expand Tree Item</span>`. 
Should read: 
`<div class="slds-tree__item slds-is-selected"><button class="slds-button slds-button_icon slds-m-right_small slds-is-disabled" disabled="" tabindex="-1" type="button"><svg aria-hidden="true" class="slds-button__icon slds-button__icon_small"><use xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"></use></svg><span class="slds-assistive-text">Expand Tree Item</span></button><span class="slds-size_1-of-1"><span class="slds-tree__item-label slds-truncate" title="Grains">Grains</span></span></div>`

**update: all tree `button` tags should have `aria-hidden="true"` on them**

**Comment by pulkonet** [comment]
@jcjuett Can you help me understand why this assistive text is needed for a disabled button, which does not intend to do the task of **expanding the tree item**? Because the item is the last node of the tree and couldn't be expanded any further, that is why the button is disabled in the first place.

This is a quick fix anyway.

**Comment by jcjuett** [comment]
@pulkonet good question. Because this is a variant, we still need it to be consistent. Although the name will not be read, it needs to be present in order to meet accessibility guidelines.

**Comment by pulkonet** [comment]
I can go ahead and add the name right away, but I think we either need a better name for the disabled button or another way to render the tree item because the button is not of any use other than providing the required margin to align with other tree items.

@jcjuett @interactivellama What are your suggestions?

**Comment by interactivellama** [comment]
Yeah, this is an odd situation and doesn't help actual user experience. We are only doing this in order to meet a technical spec and pass markup auditing. 

With that said, this is a minor update in order to pass markup audits. I like something that speaks to why it's here. I like `Presentational Disabled Button` and will get obviously flagged as a bug if it's ever not disabled. 😄 

Presentational references
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_presentation_role

@jcjuett Do you know if we add `role="presentation"` to the button wether we get then get away with not having any text?

**Comment by interactivellama** [comment]
It looks like `role=presentation` will work. https://dequeuniversity.com/rules/axe/3.1/button-name?application=axeAPI

I'd prefer that solution over adding text that is never read.

**Comment by interactivellama** [comment]
SLDS https://lightningdesignsystem.com/components/trees/ has decided go for with `aria-hidden="true"` on all the Tree buttons instead. Tree navigation is implicit in the tree for ARIA technology.

@pulkonet Please `aria-hidden="true"` to all tree buttons when it is convenient.
